packages/cli/generators/openapi/README.md

```diff
@@ -0,0 +1,165 @@
+# lb4 openapi
+
+The `openapi` command generates LoopBack 4 artifacts from an
+[OpenAPI specification](https://github.com/OAI/OpenAPI-Specification), including
+version 2.0 and 3.0.
+
+## Basic use
+
+```sh
+lb4 openapi <url or file path to the spec>
+```
+
+## Mappings
+
+We map OpenAPI operations by tag into `controllers` and schemas into TypeScript
+classes or types.
+
+### Schemas
+
+The generator first iterates through the `components.schemas` of the
+specification document and maps them into TypeScript classes or types:
+
+- Primitive types --> TypeScript type declaration
+
+```ts
+export type OrderEnum = 'ascending' | 'descending';
+```
+
+```ts
+export type Message = string;
+```
+
+- Array types --> TypeScript type declaration
+
+```ts
+import {Comment} from './comment.model';
+export type Comments = Comment[];
+```
+
+- Object type --> TypeScript class definition
+
+```ts
+import {model, property} from '@loopback/repository';
+
+/**
+ * The model class is generated from OpenAPI schema - Comment
+ */
+@model({name: 'Comment'})
+export class Comment {
+  constructor(data?: Partial<Comment>) {
+    if (data != null && typeof data === 'object') {
+      Object.assign(this, data);
+    }
+  }
+
+  @property({name: 'authorArn'})
```

Comment:
I feel the `name` attribute is redundant here because `@property` is able to infer the property name from reflection metadata.

What's the rationale for duplicating this information?

Also what happens when we change the name in one of the two places only:

```ts
@property({name: 'authorArn'})
author?: string;
```

AFAIK, we don't have any layer that would convert between typescript properties (`author`) and OpenAPI property names (`authorArn`) yet. As a result, I expect our REST layer to set extra `authorArn` property and leave `author` empty.

To me, that's another argument for letting `@property` to infer the name using reflection.

Thoughts?

Reply:
Unfortunately, some of the rest parameters have names like `my-name`, which cannot be used as a name in JavaScript. But we have to keep the original name as it's be passed in from JSON. We might have to make it as something like `@property({json: {name: 'my-name'}})` in the future.

Reply:
IIUC, you are going to convert a JSON Schema property called `my-name` to TypeScript property `myName`, is that correct? IMO, this is introducing a bug, because clients will be sending `my-name` in HTTP requests per OpenAPI description of the API, but the server will be expecting `myName` per TypeScript source.

I am proposing to handle `my-name` as follows:

```ts
class MyModel {
  @property
  'my-name': string;
}
```

Reply:
Fixed.

Reply:
Could you please update README too?